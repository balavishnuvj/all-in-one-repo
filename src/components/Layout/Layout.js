/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import Header from '../Header/Header';
import Feedback from '../Feedback';
import Footer from '../Footer';

class Layout extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    const { children } = this.props;
    return (
      <div>
        <Header />
        {children}
        <Feedback />
        <Footer />
      </div>
    );
  }
}

export default Layout;
