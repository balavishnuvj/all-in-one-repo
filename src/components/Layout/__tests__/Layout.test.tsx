/**
 * React Starter Kit (https://www.reactstarterkit.com/)
 *
 * Copyright Â© 2014-present Kriasoft, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

/* eslint-env jest */

import * as React from 'react';
import { render } from '../../../utils/shared/test/dom-test-utils';
import Layout from '../Layout';

test('renders children correctly', () => {
  // Arrange
  const { container } = render(
    <Layout>
      <div className="child" />
    </Layout>,
  );

  // Act
  // Assert
  // snapshots work great with regular DOM nodes!
  expect(container.firstChild).toMatchSnapshot();
});
