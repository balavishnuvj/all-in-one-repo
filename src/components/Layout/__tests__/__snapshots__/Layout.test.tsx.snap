// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children correctly 1`] = `
<div>
  <div
    class="sc-bdVaJa kttCLS"
  >
    <span>
      <img
        alt="logo"
      />
    </span>
    <span>
      Hello 
       !
    </span>
  </div>
  <div
    class="child"
  />
  <div
    class="root"
  >
    <div
      class="container"
    >
      <a
        class="link"
        href="https://gitter.im/kriasoft/react-starter-kit"
      >
        Ask a question
      </a>
      <span
        class="spacer"
      >
        |
      </span>
      <a
        class="link"
        href="https://github.com/kriasoft/react-starter-kit/issues/new"
      >
        Report an issue
      </a>
    </div>
  </div>
  <div
    class="root"
  >
    <div
      class="container"
    >
      <span
        class="text"
      >
        © Your Company
      </span>
      <span
        class="spacer"
      >
        ·
      </span>
      <a
        class="link"
        href="/"
      >
        Home
      </a>
      <span
        class="spacer"
      >
        ·
      </span>
      <a
        class="link"
        href="/admin"
      >
        Admin
      </a>
      <span
        class="spacer"
      >
        ·
      </span>
      <a
        class="link"
        href="/privacy"
      >
        Privacy
      </a>
      <span
        class="spacer"
      >
        ·
      </span>
      <a
        class="link"
        href="/not-found"
      >
        Not Found
      </a>
    </div>
  </div>
</div>
`;
